src/basic/env-util.c

```diff
@@ -779,7 +779,7 @@ int serialize_environment(FILE *f, char **environment) {
                 if (!ce)
                         return -ENOMEM;
 
-                fprintf(f, "env=%s\n", *e);
+                fprintf(f, "env=%s\n", ce);
```

Comment:
That is the right thing to do, but it seems that the root cause of #6152 is that [`method_set_environment`](https://github.com/systemd/systemd/blob/179e679edd5c4456da5cf09c230941d620ed4c3a/src/core/dbus-manager.c#L1602) does not unescape values before calling `strv_env_is_valid`.